components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/dao/impl/DeviceDAOImpl.java

```diff
@@ -330,4 +330,58 @@ public int getDeviceCount() throws DeviceManagementDAOException {
         return deviceCount;
     }
 
+    /**
+     * Get the list of devices that matches with the given device name.
+     *
+     * @param deviceName Name of the device.
+     * @param tenantId
+     * @return device list
+     * @throws DeviceManagementDAOException
+     */
+    @Override
+    public List<Device> getDevicesByName(String deviceName, int tenantId) throws DeviceManagementDAOException {
+        Connection conn = this.getConnection();
+        PreparedStatement stmt = null;
+        List<Device> deviceList = new ArrayList<Device>();
+        try {
+            stmt = conn.prepareStatement(
+                    "SELECT DM_DEVICE_TYPE.ID, DM_DEVICE_TYPE.NAME, DM_DEVICE.ID, DM_DEVICE.DESCRIPTION, " +
+                    "DM_DEVICE.NAME, DM_DEVICE.DATE_OF_ENROLLMENT, DM_DEVICE.DATE_OF_LAST_UPDATE, " +
+                    "DM_DEVICE.OWNERSHIP, DM_DEVICE.STATUS, DM_DEVICE.DEVICE_TYPE_ID, " +
+                    "DM_DEVICE.DEVICE_IDENTIFICATION, DM_DEVICE.OWNER, DM_DEVICE.TENANT_ID FROM " +
+                    "DM_DEVICE, DM_DEVICE_TYPE WHERE DM_DEVICE.DEVICE_TYPE_ID = DM_DEVICE_TYPE.ID " +
+                    "AND DM_DEVICE.NAME LIKE ? AND DM_DEVICE.TENANT_ID =?");
+            stmt.setString(1, deviceName + "%");
+            stmt.setInt(2, tenantId);
+            ResultSet resultSet = stmt.executeQuery();
+
+            while (resultSet.next()) {
+                Device device = new Device();
+                DeviceType deviceType = new DeviceType();
+                int id = resultSet.getInt(resultSet.getInt(1));
```

Comment:
Always use parameter names instead of indices to retrieve values out of a result set. That'd make code much clearer and robust.